### PR TITLE
Fix tenant issue in "/validate-username" and "/claims" endpoints.

### PR DESCRIPTION
**Pull request by inthirakumaaran**
fix wso2/product-is#9578

### CODE REVIEW

File: components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/ValidateUsernameApiServiceImpl.java [hunk 1]
```diff
@@ -46,25 +51,25 @@ public Response validateUsernamePost(UsernameValidationRequestDTO user) {
             return Response.status(Response.Status.BAD_REQUEST).entity("Username cannot be empty.").build();
         }
 
+        String fullyQualifiedUsername = user.getUsername();
         try {
+            String tenantDomain = resolveTenantDomain(user);
+            String username = MultitenantUtils.getTenantAwareUsername(user.getUsername());
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/ValidateUsernameApiServiceImpl.java:57]: Shall we rename this as `tenantAwareUsername`? WDYT?

**Review by somindatommy — APPROVED** [review]

File: components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/ValidateUsernameApiServiceImpl.java [hunk 2]
```diff
@@ -78,27 +83,31 @@ public Response validateUsernamePost(UsernameValidationRequestDTO user) {
...
+                        fullyQualifiedUsername, SelfRegistrationStatusCodes.CODE_USER_NAME_AVAILABLE));
                 responseDTO.setStatusCode(Integer.parseInt(SelfRegistrationStatusCodes.CODE_USER_NAME_AVAILABLE));
             }
             return Response.ok().entity(responseDTO).build();
         } catch (IdentityRecoveryException e) {
             if (LOG.isDebugEnabled()) {
-                LOG.debug("Error while checking username validity for user " + user.getUsername(), e);
+                LOG.debug("Error while checking username validity for user " + fullyQualifiedUsername, e);
+            }
+            if (e instanceof IdentityRecoveryClientException && StringUtils.isNotBlank(e.getErrorDescription())) {
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/ValidateUsernameApiServiceImpl.java:109]: Can we send a more general error client error when the description of the exception is blank? I think we can handle it via a separate catch block. WDYT?

Comment (inthirakumaaran) [reply components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/ValidateUsernameApiServiceImpl.java:109]: IMO if we send a `Response.Status.BAD_REQUEST` without proper ErrorDescription it won't be meaning full to the end user. ATST aren't we expecting an error description for all IdentityRecoveryClientExceptions? if the exception is not well defined then it is better show `Response.Status.INTERNAL_SERVER_ERROR`. WDTY?

Comment (somindatommy) [reply components/org.wso2.carbon.identity.user.endpoint/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/ValidateUsernameApiServiceImpl.java:109]: We are expecting an error message in `IdentityRecoveryClientExceptions`. But we need to handle the scenarios where the message can be blank. The purpose of the `IdentityRecoveryClientExceptions` is to represent an error in the request. `INTERNAL_SERVER_ERROR` is for errors in the server. So if we send the `INTERNAL_SERVER_ERROR` for a `BAD_REQUEST ` scenario, it conveys that server has a problem, not the request. Therefore, user will contiue to send the same request over and over again without correcting the request.

So we can send a general message such as `Invalid request` etc. WDYT?

**Review by vihanga-liyanage — APPROVED** [review]

**Pull request merged**